src/objective/rank_objective.hpp

```diff
@@ -276,6 +387,35 @@ class LambdarankNDCG : public RankingObjective {
   double max_sigmoid_input_ = 50;
   /*! \brief Factor that covert score to bin in Sigmoid table */
   double sigmoid_table_idx_factor_;
+
+  // bias correction variables
+  /*! \brief power of (click) position biases */
+  mutable std::vector<label_t> i_biases_pow_;
+
+  /*! \brief power of (unclick) position biases */
+  mutable std::vector<label_t> j_biases_pow_;
+
+  // mutable double position cost;
+  mutable std::vector<label_t> i_costs_;
+  mutable std::vector<std::vector<label_t>> i_costs_buffer_;
+
+  mutable std::vector<label_t> j_costs_;
+  mutable std::vector<std::vector<label_t>> j_costs_buffer_;
+
+  /*! 
+   * \brief Should use lambdarank with position bias correction
+   * [arxiv.org/pdf/1809.05818.pdf]
+   */
+  bool unbiased_;
+
+  /*! \brief Position bias regularizer norm */
+  double bias_p_norm_;
+
+  /*! \brief Position bias regularizer exponent, 1 / (1 + bias_p_norm_) */
+  double position_bias_regularizer;
```

Comment:
We'd better add an underscore after the member's name to be consistent with the whole code base, i.e. `position_bias_regularizer_`.